pkgs/applications/editors/neovim/utils.nix

```diff
@@ -0,0 +1,178 @@
+{ lib
+, vimUtils
+, nodejs
+, neovim-unwrapped
+, bundlerEnv
+, ruby
+, pythonPackages
+, python3Packages
+, writeText
+, wrapNeovim2
```

Comment:
What do you propose `wrapNeovimNext` ? `wrapNeovimLowLevel` (since it is intended to be less restrictive) ?

Reply:
I'm not sure. It would be best to only have one `wrapNeovim` and maybe a `generateNeovimConfiguration`, but that is probably hard to do without a hard compatibility break.

Reply:
Agreed. My proposition is to keep working on the neovim config generation and once I like and it's stable with all the features, we can start deprecating `wrapNeovim`. During the (potentially long) transition, we can't ask people to keep track of changes which is why having a compatibility-layer is good. But we still a new function to wrap neovim so I think it's fair to have it named `wrapNeovimXXXX`

Reply:
If that is the intention, maybe its best to make that explicit and call it `wrapNeovimUnstable`?

Reply:
makes sense, I like it done.